smac/intensification/successive_halving.py

```diff
@@ -482,12 +488,14 @@ def _update_stage(self, run_history: RunHistory) -> None:
             self.configs_to_run = []  # type: typing.List[Configuration]
             self.curr_inst_idx = 0
             self.running_challenger = None
-            self.curr_challengers = set()
+            self.curr_challengers = set()  # successful configs
+            self.cap_challengers = set()   # capped configs
 
         else:
             self.stage += 1
 
-            if self.stage < len(self.all_budgets):
+            if self.stage < len(self.all_budgets) and \
+                    len(self.curr_challengers) > 0:
```

Comment:
Hey, I think it would be great to actually have a unit test to check the case where all runs are capped. What do you think?

Reply:
Good idea. There is already a test case for capping. I'll extend that.